dicomutil/dicomutil.go

```diff
@@ -35,11 +37,36 @@ func main() {
 		for _, elem := range data.Elements {
 			if elem.Tag == dicomtag.PixelData {
 				data := elem.Value[0].(dicom.PixelDataInfo).Frames[0]
-				path := fmt.Sprintf("image.%d.jpg", n) // TODO: figure out the image format
+				imgType := getFormat(bytes.NewBuffer(data))
+				if imgType == "" {
+					imgType = "jpg" // set default format
+				}
+				path := fmt.Sprintf("image.%d.%s", n, imgType)
 				n++
 				ioutil.WriteFile(path, data, 0644)
 				fmt.Printf("%s: %d bytes\n", path, len(data))
 			}
 		}
 	}
 }
+
+func getFormat(file io.Reader) string {
```

Comment:
DICOM supports only JPEG, RLE and MPEG (http://dicom.nema.org/medical/dicom/current/output/html/part05.html#sect_8.2). Also, the type of the image is encoded in the TransferSyntaxUID element, and is not to be guessed from the magic header. So can you either implement more proper image-type detection, or revert the getFormat() change?